File: random-passphrase.py
import secrets
import sys

def passphrase(length = 4):
    words = open(sys.path[0] + "/dictionary.txt").read().split("\n")
    output = []
    while len(output) < length:
        word = secrets.choice(words)
        if word not in output: output.append(word)
    return " ".join(output)

print(passphrase())

File: random_passphrase.rb
def passphrase(length)
  words = File.read(File.expand_path('./dictionary.txt', __dir__)).lines.map(&:split)
  output = []
  while output.length < length
    word = words[rand(words.length)]
    output.push(word) unless output.include? word
  end
  output.join(' ')
end

puts passphrase(3)
